DR-1116 : Nightly Test Runner on Perf Github Action

snf2ye opened the pull request:
Notes for future improvements:
* We will lock the cluster w/ DR-1334
* When we can lock the cluster, we will also clean out the policies w/ DR-1339

Design questions:
* What schedule should this action run on? 
* Do we need to notify slack for any reason? (like we do on the alpha action run)
* What tests do we want to include in this run?

snf2ye committed:
wip

snf2ye committed:
first pass - update alpha to perf, switch to test runner test comand, etc

snf2ye committed:
add trigger for the new workflow, remove perf auth and comment out slack notification

snf2ye committed:
Merge branch 'develop' into sh-DR-1116-perf-action

snf2ye committed:
switch to workflow dispatch trigger

snf2ye committed:
wip# Please enter the commit message for your changes. Lines starting

snf2ye committed:
Update token

snf2ye committed:
Merge branch 'develop' into sh-DR-1116-perf-action

snf2ye committed:
Edit the new yaml

snf2ye committed:
update shell script

snf2ye committed:
move test run to new yml file

snf2ye committed:
get perf credentials to clean IAM policies

snf2ye committed:
skip IAM policy clean up for now

snf2ye committed:
Revert "skip IAM policy clean up for now"

This reverts commit 383806697d0b40ea7e4a4b43fc4f163a5f39bf5f.

snf2ye committed:
add perf role and secret id

snf2ye committed:
switch to data2 project

snf2ye committed:
add catch if gcloud fails to remeove member

snf2ye committed:
Switch to exisiting smoke test

snf2ye committed:
add gcloud configuration

snf2ye committed:
auth with dev sa

snf2ye committed:
remove glcoud config

snf2ye committed:
don't clean up IAM policies

snf2ye committed:
Merge branch 'develop' into sh-DR-1116-perf-action

snf2ye committed:
set logs to debug

snf2ye committed:
sanity check - add check for postgres

snf2ye committed:
add variables

snf2ye committed:
test while skipping kubernetes

snf2ye committed:
add back google config

snf2ye committed:
move into one action

snf2ye committed:
try whitelisting

snf2ye committed:
try adding own whitelist steps

snf2ye committed:
fix formatting

snf2ye committed:
remove unneeded check

snf2ye committed:
whitelist after configuring credentials

snf2ye committed:
move to test-dispatch file

snf2ye committed:
attempt to ditch postgres

snf2ye committed:
Add this back in with DR-1339

snf2ye committed:
Switch to new suite

snf2ye committed:
clean up

snf2ye commented on datarepo-clienttests/src/main/resources/suites/NightlyPerfWorkflow.json:
```diff
@@ -0,0 +1,6 @@
+  "name": "NightlyPerfWorkflow",
+  "description": "Performance tests to run on nightly github action",
+  "serverSpecificationFile": "perf.json",
+  "testConfigurationFiles": ["basicexamples/BasicAuthenticated.json", "basicexamples/BasicUnauthenticated.json", "basicexamples/LookupDataset.json", "profiling/BuildSnapshotWithFiles1000_50.json", "profiling/BulkLoadScale200_20.json"]
```
@marikomedlock  Are there specific tests you would want me to add to this suite?

marikomedlock replied:
Great question. I think we should discuss as a group. My opinion is that we should have at least 1 test here for each endpoint that we care about (e.g. DRS lookups, bulk file ingest, snapshot creation).

snf2ye committed:
switch back to info

snf2ye commented on .github/workflows/test-runner-on-perf.yml:
```diff
@@ -1,12 +1,91 @@
+  GOOGLE_ZONE: us-central1
+  K8_CLUSTER: jade-master-us-central1
 on:
   workflow_dispatch: {}
+  #schedule:
+  #  - cron: '0 2 * * *' # run at 2 AM UTC.
```
TODO BEFORE MERGING: switch this to a scheduled run.

RoriCremer replied:
And if we do end up using perf SAM for the perf env we should check in w QA about when we can run to avoid their testing

snf2ye replied:
Good idea!

snf2ye committed:
Merge branch 'develop' into sh-DR-1116-perf-action

snf2ye committed:
update name of action

snf2ye committed:
remove unneeded export

fboulnois commented on .github/workflows/test-runner-on-perf.yml:
```diff
@@ -1,12 +1,90 @@
+          secretId: ${{ secrets.SECRET_ID }}
+          secrets: |
+            secret/dsde/datarepo/dev/sa-key-b64 sa | B64_APPLICATION_CREDENTIALS ;
+            secret/dsde/datarepo/dev/datarepo-sql-db datarepo_db_pw | DATAREPO_PASS ;
+            secret/dsde/datarepo/dev/datarepo-sql-db stairway_db_pw | STAIRWAY_PASS ;
```
```suggestion
```

These can be removed.

fboulnois commented on .github/workflows/test-runner-on-perf.yml:
```diff
@@ -1,12 +1,90 @@
+      - name: "Checkout code"
+        uses: actions/checkout@v2
+      - name: "Import Vault dev secrets"
+        uses: hashicorp/vault-action@c8b383ee
```
```suggestion
        uses: hashicorp/vault-action@v2.0.1
```

We can probably switch this to `v2.0.1`, it looks like it was finally published.

fboulnois commented on .github/workflows/test-runner-on-perf.yml:
```diff
@@ -1,12 +1,90 @@
+          cd ${GITHUB_WORKSPACE}/${workingDir}/datarepo-client
+          echo "Building Data Repo client library"
+          ../gradlew clean assemble
+          cd ${GITHUB_WORKSPACE}/${workingDir}/datarepo-clienttests
+          export ORG_GRADLE_PROJECT_datarepoclientjar=$(ls ../datarepo-client/build/libs/*jar)
```
```suggestion
          export ORG_GRADLE_PROJECT_datarepoclientjar=$(find . -type f -name datarepo-client-*.jar -exec readlink -f {} \; )
          cd ${GITHUB_WORKSPACE}/${workingDir}/datarepo-clienttests
```

fboulnois commented on .github/test-dispatch.sh:
```diff
@@ -1,8 +1,8 @@
 #!/bin/bash
```
Does this script work now?

snf2ye replied:
Yes, it works at the moment, but It's just a testing script. We really could just delete this file before merging.

fboulnois reviewed (changes requested):
A few questions and suggestions:

snf2ye committed:
Update .github/workflows/test-runner-on-perf.yml

Co-authored-by: fboulnois <fboulnois@users.noreply.github.com>

snf2ye committed:
Update .github/workflows/test-runner-on-perf.yml

Co-authored-by: fboulnois <fboulnois@users.noreply.github.com>

snf2ye committed:
Merge branch 'develop' into sh-DR-1116-perf-action

snf2ye committed:
Switch to find command (thanks Florian!)

fboulnois reviewed (approved):
latest version lgtm!

marikomedlock commented on datarepo-clienttests/src/main/resources/suites/PRSmokeTests.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "PRSmokeTests",
-  "description": "Two smoke tests to run on PR build. One requires authentication and one does not. Only queries, no data creation.",
+  "description": "Three smoke tests to run on PR build. Two requires authentication and one does not.",
```
Nit: I think we should call this PRNightlyTests so as not to confuse the "smoke" test suite.

snf2ye replied:
I think this is the smoke test PR run. I didn't see your comments on my last PR until too late, so I thought I'd add it here. Sorry for the confusion!

marikomedlock replied:
Got it, thanks!

marikomedlock reviewed (approved):
LGTM! Thank you so much for figuring all this out.

One question: I don't see a deploy step here in the GH action. Is that correct or did I miss it somewhere? Just asking to make sure I understand. I think not having the deploy step in the GH action would make sense, so that we instead have the TestRunner re-deploy before each test config is run.

snf2ye committed:
deploy on perf

snf2ye committed:
add a few more tests to cover all endpoints

snf2ye committed:
Add maxloadarray variable to deploy

snf2ye committed:
Merge branch 'develop' into sh-DR-1116-perf-action

snf2ye committed:
final changes before merge

snf2ye commented:
@marikomedlock  
Made changes according to review:
* You're correct - there is not a deploy step in the GitHub action, but we will need to add a step that updates the perf cluster to stay in sync with the latest code changes on dev. [DR-1356] However, this was a good reminder - I switched "skipDeployment" to false so that TestRunner redeploys before each test config is run. 
* I added more tests, let me know if the list looks okay!  [Your suggestion was to have at least 1 test here for each endpoint that we care about (e.g. DRS lookups, bulk file ingest, snapshot creation)]
* One of the newly added tests was failing - setting env var DATAREPO_MAXBULKFILELOADARRAY on perf deploy fixed this issue. 

[DR-1356]: https://broadworkbench.atlassian.net/browse/DR-1356

snf2ye commented:
Int test failure documented in https://broadworkbench.atlassian.net/browse/DR-1086.

marikomedlock reviewed (approved):
LGTM
